File: OPENAPI-REST-API/swagger-client/dart/lib/model/mail_order_request.dart
part of swagger.api;

class MailOrderRequest {
  /* Mail plan service type id from getNewMail.serviceTypes (must be buyable). */
  int serviceType = null;
/* Coupon code. */
  String coupon = "";
/* Free-form note saved on the service row (used on addMail). */
  String comment = "";

  MailOrderRequest();

  @override
  String toString() {
    return 'MailOrderRequest[serviceType=$serviceType, coupon=$coupon, comment=$comment, ]';
  }

  MailOrderRequest.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    serviceType = json['serviceType'];
    coupon = json['coupon'];
    comment = json['comment'];
  }

  Map<String, dynamic> toJson() {
    return {
      'serviceType': serviceType,
      'coupon': coupon,
      'comment': comment
     };
  }

  static List<MailOrderRequest> listFromJson(List<dynamic> json) {
    return json == null ? new List<MailOrderRequest>() : json.map((value) => new MailOrderRequest.fromJson(value)).toList();
  }

  static Map<String, MailOrderRequest> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, MailOrderRequest>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new MailOrderRequest.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/MailOrderRequest.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Request body to validate (putMail) or place (addMail) a mail order. Service type ids come from getNewMail.serviceTypes.
  /// </summary>
  [DataContract]
  public class MailOrderRequest {
    /// <summary>
    /// Mail plan service type id from getNewMail.serviceTypes (must be buyable).
    /// </summary>
    /// <value>Mail plan service type id from getNewMail.serviceTypes (must be buyable).</value>
    [DataMember(Name="serviceType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "serviceType")]
    public int? ServiceType { get; set; }

    /// <summary>
    /// Coupon code.
    /// </summary>
    /// <value>Coupon code.</value>
    [DataMember(Name="coupon", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "coupon")]
    public string Coupon { get; set; }

    /// <summary>
    /// Free-form note saved on the service row (used on addMail).
    /// </summary>
    /// <value>Free-form note saved on the service row (used on addMail).</value>
    [DataMember(Name="comment", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "comment")]
    public string Comment { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class MailOrderRequest {\n");
      sb.Append("  ServiceType: ").Append(ServiceType).Append("\n");
      sb.Append("  Coupon: ").Append(Coupon).Append("\n");
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
